plutus-ledger/src/Ledger/Constraints/OffChain.hs

```diff
@@ -328,6 +331,23 @@ mkTx
     -> Either MkTxError UnbalancedTx
 mkTx lookups txc = mkSomeTx [SomeLookupsAndConstraints lookups txc]
 
+-- | Each transaction output should contain a minimum amount of Ada (this is a
+-- restriction on the real Cardano network).
+--
+-- TODO: In the future, the minimum Ada value should be configurable.
+adjustMinimumAda
+    :: ( MonadState ConstraintProcessingState m
+       )
+    => m ()
+adjustMinimumAda = do
+    unbalancedTx . tx . Tx.outputs %= (fmap adjustTxOut . filter (not . Value.isZero . txOutValue))
+
+  where
+    adjustTxOut :: TxOut -> TxOut
+    adjustTxOut txOut =
+      let missingLovelace = max 0 (Ledger.minAdaTxOut - Ada.fromValue (txOutValue txOut))
+       in txOut { txOutValue = txOutValue txOut <> Ada.toValue missingLovelace }
+
```

Comment:
Instead of doing it like that, create a function which will be called after the tx was constructed which would adjust the tx outputs to min of 2Ada. Called manually.